package/mupen64plus-rsphle/mupen64plus-rsphle.mk

```diff
@@ -5,10 +5,17 @@
 ################################################################################
 
 #MUPEN64PLUS_RSPHLE_VERSION = 5cfcc31366fc1b1f88bcbc270954b7fedcd27e21
-MUPEN64PLUS_RSPHLE_VERSION = 5d66105bda2acdfc81a3698b025c99326552146f
-MUPEN64PLUS_RSPHLE_SITE = $(call github,ricrpi,mupen64plus-rsp-hle,$(MUPEN64PLUS_RSPHLE_VERSION))
+
+ifeq ($(BR2_PACKAGE_RECALBOX_TARGET_XU4),y)
+	MUPEN64PLUS_RSPHLE_VERSION = cd9ffb7eb43bd3057f2fa58ff6268a1be9851107
+	MUPEN64PLUS_RSPHLE_SITE = $(call github,mupen64plus,mupen64plus-rsp-hle,$(MUPEN64PLUS_RSPHLE_VERSION))
+else
+	MUPEN64PLUS_RSPHLE_VERSION = 5d66105bda2acdfc81a3698b025c99326552146f
+	MUPEN64PLUS_RSPHLE_SITE = $(call github,ricrpi,mupen64plus-rsp-hle,$(MUPEN64PLUS_RSPHLE_VERSION))
+endif
+
 MUPEN64PLUS_RSPHLE_LICENSE = MIT
-MUPEN64PLUS_RSPHLE_DEPENDENCIES = sdl2 alsa-lib rpi-userland mupen64plus-core
+MUPEN64PLUS_RSPHLE_DEPENDENCIES = sdl2 alsa-lib mupen64plus-core
```

Comment:
You might add 
ifeq ($(BR2_PACKAGE_RECALBOX_TARGET_RPIX),y)
MUPEN64PLUS_RSPHLE_DEPENDENCIES += rpi-userland

Because rpi userland is still a dependencie of the rpi build. And it must be compiled before mupen.